monitor.go

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2019 The vt-go authors. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vt
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"net/url"
+)
+
+// MonitorUploader represents a  VT Monitor file upload.
+type MonitorUploader struct {
+	cli *Client
+}
+
+// Upload sends a file to VirusTotal Monitor. The file content is read from
+// the r io.Reader and sent to Monitor with the provided params.
+// The function also sends a float32 through the progress channel indicating the
+// percentage of the file that has been already uploaded. The progress channel
+// can be nil if the caller is not interested in receiving upload progress
+// updates. The received object is returned as soon as the file is uploaded.
+func (s *MonitorUploader) Upload(r io.Reader, params map[string]string, progress chan<- float32) (*Object, error) {
```

Comment:
I would say that you either need to provide a path or item id while uploading a file to Monitor. If that's the case this function shouldn't be included as part of the API because UploadFilename and UploadItem already provide what the user needs. This function can be kept for internal use.

Reply:
done https://github.com/VirusTotal/vt-go/pull/6/commits/86b97b4c98b73668301a5377f8bc973572a0ddfd